tests/generate_small_sample.py

```diff
@@ -0,0 +1,133 @@
+from test_main import TestFiles
+from pathlib import Path
+import hashlib
+
+from pysource_minimize import minimize
+import sys
+import subprocess as sp
+import textwrap
+import os
+import linecache
+from executing import Source
+from multiprocessing import Pool,get_context
+import tempfile
+import hashlib
+import time
+
+last_samples_dir = Path(__file__).parent / "last_samples"
+last_samples_dir.mkdir(exist_ok=True)
+samples_dir = Path(__file__).parent / "samples"
+
+
+# big_samples = (Path(__file__).parent / "samples").glob("*.py")
+# big_samples = list(Path.home().rglob("*.py"))
+small_samples = Path(__file__).parent / "small_samples"
+
+
+def big_samples():
+    yield from last_samples_dir.rglob("*.py")
+    yield from samples_dir.rglob("*.py")
+
+    hashes = set()
+    root = Path.home()
+    # root = Path(__file__).parent / "samples"
+    for p in root.rglob("*.py"):
+        try:
+            content = p.read_text()
+        except:
+            continue
+
+        if content.count("\n") > 100000:
+            continue
+
+        h = hashlib.sha256(content.encode()).hexdigest()
+        if h in hashes:
+            continue
+        hashes.add(h)
+        yield p
+
+
+def test_file(filename: Path):
+    sys.setrecursionlimit(3000)
+
+    if filename.read_text().count("\n") > 50000:
+        return True, filename
+
+    # clear caches
+    linecache.clearcache()
+    for cache_name in ("__source_cache_with_lines", "__executing_cache"):
+        if hasattr(Source, cache_name):
+            delattr(Source, cache_name)
+
+    test = TestFiles()
+    try:
+        test.check_filename(filename, check_names=True)
+    except:
+        return False, filename
+
+    return True, filename
```

Comment:
Returning the filename each time looks pretty weird.

Reply:
handy for multiprocessing.Pool.imap_unordered.

Reply:
Yes, but I'd prefer for this function to just return the boolean because that's its responsibility. A separate utility function or something can zip with the inputs.